src/pages/samples/reversedZ.ts

```diff
@@ -0,0 +1,1008 @@
+void main() {
+}
+`,
+
+  vertexPrecisionErrorPass: `#version 450
+layout(set = 0, binding = 0) uniform Uniforms {
+  mat4 modelMatrix[${numInstances}];
+} uniforms;
+
+layout(set = 0, binding = 1) uniform CameraMatrix {
+  mat4 viewProjectionMatrix;
+} camera;
+
+
+layout(location = 0) in vec4 position;
+
+layout(location = 0) out vec4 clipPos;
+
+void main() {
+  gl_Position = camera.viewProjectionMatrix * uniforms.modelMatrix[gl_InstanceIndex] * position;
+  clipPos = gl_Position;
+}
+`,
+
+  fragmentPrecisionErrorPass: `#version 450
+layout(location = 0) in vec4 clipPos;
+layout(location = 0) out vec4 outColor;
+
+layout(set = 1, binding = 0) uniform texture2D depthMap;
+layout(set = 1, binding = 1) uniform sampler depthSampler;
+
+void main() {
+  vec2 fragUV = gl_FragCoord.xy / vec2(${kDefaultCanvasWidth.toFixed(
+    1
+  )}, ${kDefaultCanvasHeight.toFixed(1)});
+  float depthValue = texture(sampler2D(depthMap, depthSampler), fragUV).r;
+  float error = abs(clipPos.z / clipPos.w - depthValue);
+  outColor = vec4(vec3(error * 2000000.0), 1.0);
+}
+`,
+
+  vertexTextureQuad: `#version 450
+const vec2 pos[6] = vec2[6](
+  vec2(-1.0f, -1.0f), vec2(1.0f, -1.0f), vec2(-1.0f, 1.0f),
+  vec2(-1.0f, 1.0f), vec2(1.0f, -1.0f), vec2(1.0f, 1.0f)
+);
+void main() {
+  gl_Position = vec4(pos[gl_VertexIndex], 0.0, 1.0);
+}
+  `,
+  fragmentTextureQuad: `#version 450
+layout(set = 0, binding = 0) uniform texture2D map;
+layout(set = 0, binding = 1) uniform sampler mapSampler;
+
+layout(location = 0) out vec4 outColor;
+
+void main() {
+  vec2 fragUV = gl_FragCoord.xy / vec2(${kDefaultCanvasWidth.toFixed(
+    1
+  )}, ${kDefaultCanvasHeight.toFixed(1)});
+  float value = texture(sampler2D(map, mapSampler), fragUV).r;
+  outColor = vec4(vec3(value), 1.0);
+}
+  `,
+};
+
+const wgslShaders = {
+  vertex: `
+[[block]] struct Uniforms {
+  [[offset(0)]] modelMatrix : [[stride(${matrixStride})]] array<mat4x4<f32>, ${numInstances}>;
+};
+[[block]] struct Camera {
+  [[offset(0)]] viewProjectionMatrix : mat4x4<f32>;
+};
+
+[[builtin(instance_index)]] var<in> instanceIdx : i32;
+[[binding(0), group(0)]] var<uniform> uniforms : Uniforms;
+[[binding(1), group(0)]] var<uniform> camera : Camera;
+
+[[location(0)]] var<in> position : vec4<f32>;
+[[location(1)]] var<in> color : vec4<f32>;
+
+[[builtin(position)]] var<out> Position : vec4<f32>;
+[[location(0)]] var<out> fragColor : vec4<f32>;
+
+[[stage(vertex)]]
+fn main() -> void {
+  Position = camera.viewProjectionMatrix * uniforms.modelMatrix[instanceIdx] * position;
+  fragColor = color;
+}
+`,
+  fragment: `
+[[location(0)]] var<in> fragColor : vec4<f32>;
+[[location(0)]] var<out> outColor : vec4<f32>;
+
+[[stage(fragment)]]
+fn main() -> void {
+  outColor = fragColor;
+}
+`,
+  vertexDepthPrePass: `
+[[block]] struct Uniforms {
+  [[offset(0)]] modelMatrix : [[stride(${matrixStride})]] array<mat4x4<f32>, ${numInstances}>;
+};
+[[block]] struct Camera {
+  [[offset(0)]] viewProjectionMatrix : mat4x4<f32>;
```

Comment:
```suggestion
  viewProjectionMatrix : mat4x4<f32>;
```
offset shouldn't be required anymore